### PR TITLE
RELATED: RAIL-3492 save and save as fixes

### PR DESCRIPTION
**Pull request by no23reason**
* add saving slice and use it to disable the save button in Save as dialog while the saving is in progress
* add CommandStarted event and use it to track start of save operations
* fix logic displaying the tooltip for the Save as button

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/saveAsDashboardHandler.ts [hunk 1]
```diff
@@ -132,28 +134,36 @@ export function* saveAsDashboardHandler(
...
-    const {
-        payload: { switchToCopy },
-    } = cmd;
-    const result: SagaReturnType<typeof saveAs> = yield call(saveAs, ctx, saveAsCtx);
-    const { dashboard, batch } = result;
-
-    if (batch) {
-        yield put(batch);
+    try {
+        yield put(dashboardCommandStarted(ctx, cmd));
```
Comment (lupko) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/saveAsDashboardHandler.ts:138]: I think this is better in the rootCommandHandler saga. let's emit these events for all commands right away, then plan for a task to clean up the duplicate *Requested events.

File: libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/saveDashboardHandler.ts [hunk 2]
```diff
@@ -170,31 +171,39 @@ export function* saveDashboardHandler(
...
-
-    if (isNewDashboard) {
-        yield setContext({
-            dashboardContext: {
-                ...ctx,
-                dashboardRef: dashboard.ref,
-            },
-        });
+    try {
+        yield put(dashboardCommandStarted(ctx, cmd));
```
Comment (lupko) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/saveDashboardHandler.ts:175]: same as saveAs - perhaps better in root command handler?

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5145/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6847/ : FAILURE in 6m 11s

File: libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/saveAsDashboardHandler.ts (hunk 1 above)
Comment (no23reason) [reply libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/saveAsDashboardHandler.ts]: Good point, fixed in 0652b73

File: libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/saveDashboardHandler.ts (hunk 2 above)
Comment (no23reason) [reply libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/saveDashboardHandler.ts]: Good point, fixed in 0652b73

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by lupko — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5149/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6851/ : FAILURE in 17m 16s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1733

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1472/ : SUCCESS in 59s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1749/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/535/ : SUCCESS in 12m 01s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6854/ : FAILURE in 14m 48s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5154/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6859/ : SUCCESS in 17m 37s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1733

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1475/ : SUCCESS in 59s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1752/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/538/ : SUCCESS in 18m 23s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6860/ : SUCCESS in 24m 46s

**Pull request merged**